cellarium/cas/postprocessing/cell_ontology/cell_ontology_cache.py

```diff
@@ -0,0 +1,142 @@
+
+# only keep nodes with the following prefix when parsing CL ontology
+CL_PREFIX = "CL_"
+
+# the 'eukaryotic cell' node
+CL_CELL_ROOT_NODE = "CL_0000255"
+
+
+class CellOntologyCache:
+    """
+    A class representing a cache for the Cell Ontology (CL).
+
+    Args:
+        cl_owl_path (str): The path to the Cell Ontology OWL file. Defaults to DEFAULT_CL_OWL_PATH.
+
+    Attributes:
+        cl_classes (list): A list of Cell Ontology classes with a singleton label.
+        cl_names (list): A list of names of Cell Ontology classes.
+        cl_labels (list): A list of labels of Cell Ontology classes.
+        cl_classes_set (set): A set of Cell Ontology classes.
+        cl_names_to_labels_map (dict): A mapping of Cell Ontology class names to labels.
+        cl_labels_to_names_map (dict): A mapping of Cell Ontology class labels to names.
+        cl_names_to_idx_map (dict): A mapping of Cell Ontology class names to indices.
+        cl_idx_to_names_map (dict): A mapping of indices to Cell Ontology class names.
+        cl_graph (nx.DiGraph): A networkx graph representing the Cell Ontology.
+
+    """
+
+    def __init__(self, cl_owl_path: str = DEFAULT_CL_OWL_PATH):
+        """
+        Initialize the CellOntologyCache object.
+
+        Loads the Cell Ontology from the specified OWL file, filters out classes with a singleton label,
+        and builds a networkx graph representing the Cell Ontology.
+
+        :param cl_owl_path: The path to the Cell Ontology OWL file. Defaults to DEFAULT_CL_OWL_PATH.
+        :type cl_owl_path: str
+
+        """
+
+        with suppress_stderr():
+            log(logging.INFO, f"Loading cell ontology OWL from:\n{cl_owl_path}")
+            cl = owlready2.get_ontology(cl_owl_path).load()
+
+        # only keep CL classes with a singleton label
+        cl_classes = list(
+            _class for _class in cl.classes() if _class.name.startswith(CL_PREFIX) and len(_class.label) == 1
+        )
+
+        cl_names = [_class.name for _class in cl_classes]
+        cl_labels = [_class.label[0] for _class in cl_classes]
+        assert len(set(cl_names)) == len(cl_classes)
+        assert len(set(cl_labels)) == len(cl_classes)
+
+        cl_classes_set = set(cl_classes)
+        cl_names_to_labels_map = {cl_name: cl_label for cl_name, cl_label in zip(cl_names, cl_labels)}
+        cl_labels_to_names_map = {cl_label: cl_name for cl_name, cl_label in zip(cl_names, cl_labels)}
+        cl_names_to_idx_map = {cl_name: idx for idx, cl_name in enumerate(cl_names)}
+        cl_idx_to_names_map = {idx: cl_name for idx, cl_name in enumerate(cl_names)}
+
+        # build a networkx graph from CL
+        cl_graph = nx.DiGraph(name="CL graph")
+
+        for cl_class in cl_classes:
+            cl_graph.add_node(cl_class.name)
+
+        for self_cl_class in cl_classes:
+            for parent_cl_class in cl.get_parents_of(self_cl_class):
+                if parent_cl_class not in cl_classes_set:
```

Comment:
Extremely nitpicky, but would it be very slightly better to do:
```python
if parent_cl_class in cl_classes_set:
    cl_graph.add_edge(parent_cl_class.name, self_cl_class.name)
```

Reply:
Hmm...I'll look more closely...would that not reverse the logic?